app/controllers/application_controller.rb

```diff
@@ -2,4 +2,24 @@ class ApplicationController < ActionController::Base
   # Prevent CSRF attacks by raising an exception.
   # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :exception
+
+  private
+
+  def current_user
+    @_current_user ||= session[:user_id]
```

Comment:
whenever `current_user` gets called, the @_current_user will get conditionally defined but the User.find will always get run because it's never cached anywhere, nor served out of a cache.